CMakeLists.txt

```diff
@@ -48,75 +33,87 @@ LINK_DIRECTORIES(
         /usr/lib
 )
 
-AUX_SOURCE_DIRECTORY(trantor/utils DIR_SRC)
-AUX_SOURCE_DIRECTORY(trantor/net DIR_SRC)
-AUX_SOURCE_DIRECTORY(trantor/net/inner DIR_SRC)
-find_package (OpenSSL)
-if(OpenSSL_FOUND)
-    #add_definitions(-DUSE_OPENSSL)
-    set(DEFS "USE_OPENSSL;${DEFS}")
+SET(public_net_headers
+    trantor/net/EventLoop.h
+    trantor/net/EventLoopThread.h
+    trantor/net/EventLoopThreadPool.h
+    trantor/net/InetAddress.h
+    trantor/net/TcpClient.h
+    trantor/net/TcpConnection.h
+    trantor/net/TcpServer.h
+    trantor/net/callbacks.h)
+    
+SET(public_utils_headers
+    trantor/utils/AsyncFileLogger.h
+    trantor/utils/ConcurrentTaskQueue.h
+    trantor/utils/Date.h
+    trantor/utils/Funcs.h
+    trantor/utils/LockFreeQueue.h
+    trantor/utils/LogStream.h
+    trantor/utils/Logger.h
+    trantor/utils/MsgBuffer.h
+    trantor/utils/NonCopyable.h
+    trantor/utils/ObjectPool.h
+    trantor/utils/SerialTaskQueue.h
+    trantor/utils/TaskQueue.h
+    trantor/utils/TimingWheel.h)
+
+SOURCE_GROUP( "Public API" FILES ${public_net_headers} ${public_utils_headers})
+
+SET(trantor_sources
+    trantor/utils/AsyncFileLogger.cc
+    trantor/utils/ConcurrentTaskQueue.cc
+    trantor/utils/Date.cc
+    trantor/utils/LogStream.cc
+    trantor/utils/Logger.cc
+    trantor/utils/MsgBuffer.cc
+    trantor/utils/SerialTaskQueue.cc
+    trantor/utils/TimingWheel.cc
+    trantor/net/EventLoop.cc
+    trantor/net/EventLoopThread.cc
+    trantor/net/EventLoopThreadPool.cc
+    trantor/net/InetAddress.cc
+    trantor/net/TcpClient.cc
+    trantor/net/TcpServer.cc
+    trantor/net/inner/Acceptor.cc
+    trantor/net/inner/Channel.cc
+    trantor/net/inner/Connector.cc
+    trantor/net/inner/Poller.cc
+    trantor/net/inner/Socket.cc
+    trantor/net/inner/TcpConnectionImpl.cc
+    trantor/net/inner/Timer.cc
+    trantor/net/inner/TimerQueue.cc)
+
+FIND_PACKAGE(OpenSSL)
+IF(OpenSSL_FOUND)
+    add_definitions(-DUSE_OPENSSL)
```

Comment:
Although this is entirely fine, but you could conditionally include a `SSLConnectionSkipped.cc` file instead, that exposes the same API but with no-ops (or errors) if called.  This is really what you want to do too if the API is exposed to users as you don't want the API changing depending on compiled-in features (otherwise 'that' should be another side library to optionally link in instead).

Reply:
I added a `SSLConnectionSkipped.cc` file and removed the `USE_OPENSSL`